modules/tinymce/src/core/main/ts/dom/TrimHtml.ts

```diff
@@ -5,13 +5,96 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
+import { Obj, Strings, Type } from '@ephox/katamari';
+
 import DomSerializer from '../api/dom/Serializer';
-import SaxParser from '../api/html/SaxParser';
+import Schema from '../api/html/Schema';
 import * as Zwsp from '../text/Zwsp';
 
+const isConditionalComment = (html: string, startIndex: number) => /^\s*\[if [\w\W]+\]>.*<!\[endif\](--!?)?>/.test(html.substr(startIndex));
+
+const findCommentEndIndex = (html: string, isBogus: boolean, startIndex: number = 0) => {
+  const lcHtml = html.toLowerCase();
+  if (lcHtml.indexOf('[if ', startIndex) !== -1 && isConditionalComment(lcHtml, startIndex)) {
+    const endIfIndex = lcHtml.indexOf('[endif]', startIndex);
+    return lcHtml.indexOf('>', endIfIndex);
+  } else {
+    if (isBogus) {
+      const endIndex = lcHtml.indexOf('>', startIndex);
+      return endIndex !== -1 ? endIndex : lcHtml.length;
+    } else {
+      const endCommentRegexp = /--!?>/g;
+      endCommentRegexp.lastIndex = startIndex;
+      const match = endCommentRegexp.exec(html);
+      return match ? match.index + match[0].length : lcHtml.length;
+    }
+  }
+};
+
+/**
+ * Returns the index of the matching end tag for a specific start tag. This can
+ * be used to skip all children of a parent element from being processed.
+ *
+ * @private
```

Comment:
Does this still need the `@private`?

Reply:
I believe so since it's formatted as a JS doc string. I'm not convinced it should be but the same applies to where it was previously so for safety I just left it as is.

Reply:
Have converted it to just a regular command and I also realised I needed to port the tests for this function from `SaxParserTest.ts` so have done that as well.